compiler/crates/relay-lsp/src/rename/mod.rs

```diff
@@ -0,0 +1,413 @@
+                        SourceLocationKey::embedded(new_file_uri.as_ref(), index),
+                    )
+                    .item;
+
+                    for definition in &document.definitions {
+                        let changes = match definition {
+                            ExecutableDefinition::Fragment(frag_def) => {
+                                let old_frag_name = frag_def.name.value;
+                                let new_frag_name = replace_prefix(
+                                    &old_frag_name.to_string(),
+                                    &old_name_prefix,
+                                    &new_name_prefix,
+                                );
+
+                                rename_fragment(old_frag_name, new_frag_name, program, root_dir)?
+                            }
+                            ExecutableDefinition::Operation(OperationDefinition {
+                                name: Some(operation_name_identifier),
+                                ..
+                            }) => {
+                                let new_operation_name = replace_prefix(
+                                    &operation_name_identifier.value.to_string(),
+                                    &old_name_prefix,
+                                    &new_name_prefix,
+                                );
+
+                                let name_range =
+                                    text_source.to_span_range(operation_name_identifier.span);
+
+                                let location = LspLocation::new(old_file_uri.clone(), name_range);
+
+                                rename_operation(new_operation_name, location)
+                            }
+                            ExecutableDefinition::Operation(
+                                graphql_syntax::OperationDefinition { name: None, .. },
+                            ) => HashMap::new(),
+                        };
+
+                        merge_text_changes(&mut rename_changes, changes);
+                    }
+                }
+                _ => (),
+            };
+        }
+    }
+
+    Ok(Some(WorkspaceEdit {
+        changes: Some(rename_changes),
+        ..Default::default()
+    }))
+}
+
+fn replace_prefix(s: &str, old_prefix: &str, new_prefix: &str) -> String {
+    if let Some(rest) = s.strip_prefix(old_prefix) {
+        let mut result = String::from(new_prefix);
+        result.push_str(rest);
+        return result;
+    }
+
+    s.to_string()
+}
+
+fn rename_relay_resolver_field(
+    field_name: StringKey,
+    type_name: StringKey,
+    new_field_name: &str,
+    program: &Program,
+    root_dir: &PathBuf,
+) -> LSPRuntimeResult<HashMap<Url, Vec<TextEdit>>> {
+    let mut changes = HashMap::<Url, Vec<TextEdit>>::new();
+
+    for location in find_field_locations(program, field_name, type_name).unwrap_or_else(|| vec![]) {
+        let lsp_location = transform_relay_location_to_lsp_location(root_dir, location)?;
```

Comment:
Internally, we may have a huge number of items here and below in `rename_fragment`. I wonder if it would improve performance to tackle these in parallel (par_iter).